runtime error on "blobfuse2 mount all"

### Which version of blobfuse was used?
2

### Which OS distribution and version are you using?
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

running through WSL

### If relevant, please share your mount command.
blobfuse2 mount all /mnt/azstorage



### What was the issue encountered?
runtime error
```
panic: runtime error: index out of range [4] with length 4

goroutine 1 [running]:
github.com/Azure/azure-storage-fuse/v2/cmd.mountAllContainers(0xc00009e960, 0x2, 0x2, 0x0, 0x0, 0x7ffe7fbe3956, 0xe, 0x0, 0xe79982, 0x9)
        /mnt/vss/_work/1/s/azure-storage-fuse/cmd/mount_all.go:321 +0xf12
github.com/Azure/azure-storage-fuse/v2/cmd.processCommand(0x0, 0x0)
        /mnt/vss/_work/1/s/azure-storage-fuse/cmd/mount_all.go:180 +0x599
github.com/Azure/azure-storage-fuse/v2/cmd.glob..func9(0x149b920, 0xc00034b270, 0x1, 0x1, 0x0, 0x0)
        /mnt/vss/_work/1/s/azure-storage-fuse/cmd/mount_all.go:82 +0x9f
github.com/spf13/cobra.(*Command).execute(0x149b920, 0xc00034b250, 0x1, 0x1, 0x149b920, 0xc00034b250)
        /home/cloudtest/go/pkg/mod/github.com/gapra-msft/cobra@v1.4.1-0.20220411185530-5b83e8ba06dd/command.go:882 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x149c100, 0x3, 0x3, 0xc00040a7c0)
        /home/cloudtest/go/pkg/mod/github.com/gapra-msft/cobra@v1.4.1-0.20220411185530-5b83e8ba06dd/command.go:1000 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/cloudtest/go/pkg/mod/github.com/gapra-msft/cobra@v1.4.1-0.20220411185530-5b83e8ba06dd/command.go:928
github.com/Azure/azure-storage-fuse/v2/cmd.Execute(0xffffffff, 0xc000042118)
        /mnt/vss/_work/1/s/azure-storage-fuse/cmd/root.go:262 +0x8d
main.main()
        /mnt/vss/_work/1/s/azure-storage-fuse/main.go:45 +0x25
```



### Have you found a mitigation/solution?
I looked at the code and it seems that the logic is expecting there to be 5 cli parms when there is only 4 in my command here...
https://github.com/Azure/azure-storage-fuse/blob/main/cmd/mount_all.go#L321

So I added an innocuous param (for me at least) `--disable-version-check=true`  and it seems to work now. 

I would expect an error message of some kind if it is missing a required parm rather than a runtime error 

### Please share logs if available.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 - [#1002](https://github.com/Azure/azure-storage-fuse/issues/1002) In case version check fails to connect to public container, dump a log to check network and proxy settings.
 - [#1006](https://github.com/Azure/azure-storage-fuse/issues/1006) Remove user and group config from logrotate file.
 - [csi-driver #809](https://github.com/kubernetes-sigs/blob-csi-driver/issues/809) Fail to mount when uid/gid are provided on command line.
+- [#1032](https://github.com/Azure/azure-storage-fuse/issues/1032) CLI params updation fix in `mount all` command.
 
 ## 2.0.1 (2022-12-02)
 - Copy of GA release of Blobfuse2. This release was necessary to ensure the GA version resolves ahead of the preview versions.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,7 @@
 - [#1002](https://github.com/Azure/azure-storage-fuse/issues/1002) In case version check fails to connect to public container, dump a log to check network and proxy settings.
 - [#1006](https://github.com/Azure/azure-storage-fuse/issues/1006) Remove user and group config from logrotate file.
 - [csi-driver #809](https://github.com/kubernetes-sigs/blob-csi-driver/issues/809) Fail to mount when uid/gid are provided on command line.
-- [#1032](https://github.com/Azure/azure-storage-fuse/issues/1032) CLI params updation fix in `mount all` command.
+- [#1032](https://github.com/Azure/azure-storage-fuse/issues/1032) `mount all` CLI params parsing fix when no `config-file` and `tmp-path` is provided.
 
 ## 2.0.1 (2022-12-02)
 - Copy of GA release of Blobfuse2. This release was necessary to ensure the GA version resolves ahead of the preview versions.

--- a/cmd/mount_all.go
+++ b/cmd/mount_all.go
@@ -317,8 +317,8 @@ func mountAllContainers(containerList []string, configFile string, mountPath str
 			cliParams[2] = "--config-file=" + contConfigFile
 		} else {
 			cliParams[2] = "--foreground=false"
-			cliParams = append(cliParams, "--container-name="+container)
-			cliParams = append(cliParams, "--tmp-path="+filepath.Join(fileCachePath, container))
+			updateCliParams(&cliParams, "container-name", container)
+			updateCliParams(&cliParams, "tmp-path", filepath.Join(fileCachePath, container))
 		}
 
 		// Now that we have mount path and config file for this container fire a mount command for this one
@@ -338,6 +338,16 @@ func mountAllContainers(containerList []string, configFile string, mountPath str
 
 	fmt.Printf("%d of %d containers were successfully mounted\n", (len(containerList) - failCount), len(containerList))
 	return nil
+}
+
+func updateCliParams(cliParams *[]string, key string, val string) {
+	for i := 3; i < len(*cliParams); i++ {
+		if strings.Contains((*cliParams)[i], "--"+key) {
+			(*cliParams)[i] = "--" + key + "=" + val
+			return
+		}
+	}
+	*cliParams = append(*cliParams, "--"+key+"="+val)
 }
 
 func writeConfigFile(contConfigFile string) error {

--- a/cmd/mount_all.go
+++ b/cmd/mount_all.go
@@ -317,8 +317,8 @@ func mountAllContainers(containerList []string, configFile string, mountPath str
 			cliParams[2] = "--config-file=" + contConfigFile
 		} else {
 			cliParams[2] = "--foreground=false"
-			cliParams[3] = "--container-name=" + container
-			cliParams[4] = "--tmp-path=" + filepath.Join(fileCachePath, container)
+			cliParams = append(cliParams, "--container-name="+container)
+			cliParams = append(cliParams, "--tmp-path="+filepath.Join(fileCachePath, container))
 		}
 
 		// Now that we have mount path and config file for this container fire a mount command for this one

--- a/cmd/mount_test.go
+++ b/cmd/mount_test.go
@@ -427,6 +427,26 @@ func (suite *mountTestSuite) TestFuseOptions() {
 	}
 }
 
+func (suite *mountTestSuite) TestUpdateCliParams() {
+	defer suite.cleanupTest()
+
+	cliParams := []string{"blobfuse2", "mount", "~/mntdir/", "--foreground=false"}
+
+	updateCliParams(&cliParams, "tmp-path", "tmpPath1")
+	suite.assert.Equal(len(cliParams), 5)
+	suite.assert.Equal(cliParams[4], "--tmp-path=tmpPath1")
+
+	updateCliParams(&cliParams, "container-name", "testCnt1")
+	suite.assert.Equal(len(cliParams), 6)
+	suite.assert.Equal(cliParams[5], "--container-name=testCnt1")
+
+	updateCliParams(&cliParams, "tmp-path", "tmpPath2")
+	updateCliParams(&cliParams, "container-name", "testCnt2")
+	suite.assert.Equal(len(cliParams), 6)
+	suite.assert.Equal(cliParams[4], "--tmp-path=tmpPath2")
+	suite.assert.Equal(cliParams[5], "--container-name=testCnt2")
+}
+
 func TestMountCommand(t *testing.T) {
 	confFile, err := ioutil.TempFile("", "conf*.yaml")
 	if err != nil {

--- a/test/mount_test/mount_test.go
+++ b/test/mount_test/mount_test.go
@@ -243,9 +243,9 @@ func (suite *mountSuite) TestEnvVarMount() {
 	os.Setenv("AZURE_STORAGE_ACCOUNT_CONTAINER", viper.GetString("azstorage.container"))
 	os.Setenv("AZURE_STORAGE_ACCOUNT_TYPE", viper.GetString("azstorage.type"))
 
-	tempFile := viper.GetString("file_cache.path")
+	tempCachePath := viper.GetString("file_cache.path")
 
-	mountCmd := exec.Command(blobfuseBinary, "mount", mntDir, "--tmp-path="+tempFile)
+	mountCmd := exec.Command(blobfuseBinary, "mount", mntDir, "--tmp-path="+tempCachePath)
 	cliOut, err := mountCmd.Output()
 	fmt.Println(string(cliOut))
 	suite.Equal(0, len(cliOut))
@@ -261,6 +261,35 @@ func (suite *mountSuite) TestEnvVarMount() {
 
 	// unmount
 	blobfuseUnmount(suite, mntDir)
+
+	mountAllCmd := exec.Command(blobfuseBinary, "mount", "all", mntDir, "--tmp-path="+tempCachePath)
+	cliOut, err = mountAllCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.NotEqual(0, len(cliOut))
+	suite.Equal(nil, err)
+
+	// wait for mount
+	time.Sleep(10 * time.Second)
+
+	// list blobfuse mounted directories
+	cliOut = listBlobfuseMounts(suite)
+	suite.NotEqual(0, len(cliOut))
+	suite.Contains(string(cliOut), mntDir)
+
+	// unmount
+	blobfuseUnmount(suite, mntDir)
+
+	err = os.RemoveAll(mntDir)
+	suite.Equal(nil, err)
+
+	err = os.RemoveAll(tempCachePath)
+	suite.Equal(nil, err)
+
+	err = os.Mkdir(mntDir, 0777)
+	suite.Equal(nil, err)
+
+	err = os.Mkdir(tempCachePath, 0777)
+	suite.Equal(nil, err)
 
 	os.Unsetenv("AZURE_STORAGE_ACCOUNT")
 	os.Unsetenv("AZURE_STORAGE_ACCESS_KEY")

